asynq/scheduler.py

```diff
@@ -574,6 +575,9 @@ def extract_futures(value, result):
     elif type(value) is tuple or type(value) is list:
         for item in value:
             extract_futures(item, result)
+    elif type(value) is dict:
+        for value in six.iteritems(value):
```

Comment:
I think that re-using `value` as a loop variable will shadow the initial `value` in Python 2. Also, `iteritems` returns both the key and the value as a tuple and I believe you need only the value here, right?

Perhaps you should consider the following:

```
for item in six.itervalues(value):
    extract_futures(item, result)
```